### PR TITLE
Uar 372 sumary view reviewed mo

### PR DESCRIPTION
**Pull request by cdrainCH**
### JIRA link

https://companieshouse.atlassian.net/jira/software/c/projects/UAR/boards/531?modal=detail&selectedIssue=UAR-372&sprint=2866

### Change description

Adds reviewed managing officers to the reviewed Bo/Mo table on Bo/Mo summary page

### Work checklist

- [ ] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/controllers/update/beneficial.owner.type.controller.ts [hunk 1]
```diff
@@ -42,16 +44,24 @@ export const get = async (req: Request, res: Response, next: NextFunction) => {
...
-    const hasExistingBos = allBos.find(bo => bo.ch_reference) !== undefined;
     const hasNewlyAddedBos = allBos.find(bo => bo.ch_reference === undefined) !== undefined;
 
+    const allBosMos = [
+      ...allBos,
+      ...(appData[ManagingOfficerCorporateKey] ?? []),
+      ...(appData[ManagingOfficerKey] ?? [])
+    ];
+
+    const hasExistingBosMos = allBosMos.find(boMo => boMo.ch_reference) !== undefined;
```
Comment (AndreKramer7654) [inline src/controllers/update/beneficial.owner.type.controller.ts:58]: Suggest some() instead of find()

Comment (AndreKramer7654) [reply src/controllers/update/beneficial.owner.type.controller.ts:58]: also would be explicit about ch_reference !== undefined

File: src/routes/index.ts [hunk 2]
```diff
@@ -533,6 +533,17 @@ router.route(config.UPDATE_MANAGING_OFFICER_URL + config.ID)
...
+router.route(config.UPDATE_REVIEW_INDIVIDUAL_MANAGING_OFFICER_URL)
+  .all(
+    authentication,
+    companyAuthentication,
+    navigation.hasUpdatePresenter
+  )
+  .get(updateReviewIndividualManagingOfficer.get)
+  .post(...validator.updateBeneficialOwnerAndManagingOfficerType, checkValidations, updateReviewIndividualManagingOfficer.post);
+
+router.get(config.UPDATE_REVIEW_BENEFICIAL_OWNER_GOV_URL + config.UPDATE_REVIEW_OWNERS_PARAMS, authentication, navigation.hasUpdatePresenter, updateReviewBeneficialOwnerGov.get);
```
Comment (AndreKramer7654) [inline src/routes/index.ts:545]: Not sure we needed GOV now but ok. Could you use mulit-line formatting as in lines 536-545

File: src/utils/update/mapper.utils.ts [hunk 3]
```diff
@@ -98,5 +98,9 @@ type AddressMatches = {
 };
 
 export const isSameAddress: AddressMatches = (address1: any, address2?: any) => {
-  return !address2 || Object.keys(address1).every(key => address1[key] === address2[key]);
```
Comment (AndreKramer7654) [inline src/utils/update/mapper.utils.ts:101]: please check existing usage of this. I think it was saying address2 being undefined - defaults address2 to address1 so they are equal. So address2 null => the addresses are equal!

File: src/utils/update/review.beneficial.owner.ts [hunk 4]
```diff
@@ -38,9 +38,12 @@ const checkBoIndividualValidation = (boi: BeneficialOwnerIndividual): boolean =>
   return boi?.usual_residential_address ? true : false;
 };
 
-const checkBoValidation = (bo: BeneficialOwnerGov): boolean => {
-  const principalAddress = bo.principal_address || {};
-  return Object.keys(principalAddress).length ? true : false;
+const checkBoOtherValidation = (boOther: BeneficialOwnerOther): boolean => {
+  return boOther?.principal_address ? true : false;
```
Comment (AndreKramer7654) [inline src/utils/update/review.beneficial.owner.ts:42]: What if principle_address is {}?

File: src/controllers/update/beneficial.owner.type.controller.ts (hunk 1 above)
Comment (cdrainCH) [reply src/controllers/update/beneficial.owner.type.controller.ts:58]: Functionality works with find() and it was the approach pushed for the last ticket

File: src/utils/update/mapper.utils.ts (hunk 3 above)
Comment (cdrainCH) [reply src/utils/update/mapper.utils.ts:101]: I don't believe this was working as expected. The "yes" radio button was being selected despite having a service address -  change has fixed this

File: src/routes/index.ts (hunk 2 above)
Comment (cdrainCH) [reply src/routes/index.ts]: Done.

File: src/controllers/update/beneficial.owner.type.controller.ts [hunk 5]
```diff
@@ -116,6 +127,7 @@ const retrieveManagingOfficers = async (req: Request, appData: ApplicationData)
         } else if (officer.officerRole === "corporate-managing-officer") {
           const managingOfficerCorporate = mapToManagingOfficerCorporate(officer);
           logger.info("Loaded Corporate Managing Officer " + managingOfficerCorporate.id + " is " + managingOfficerCorporate.name);
+          appData.update?.review_managing_officers_corporate?.push(managingOfficerCorporate);
```
Comment (stephanie-ch) [inline src/controllers/update/beneficial.owner.type.controller.ts:130]: Do we need this? Reckon it's part of the next review ticket

File: src/routes/index.ts [hunk 6]
```diff
@@ -533,6 +554,22 @@ router.route(config.UPDATE_MANAGING_OFFICER_URL + config.ID)
...
+router.route(config.UPDATE_REVIEW_INDIVIDUAL_MANAGING_OFFICER_URL)
+  .all(
+    authentication,
+    companyAuthentication,
+    navigation.hasUpdatePresenter
+  )
+  .get(updateReviewIndividualManagingOfficer.get)
+  .post(...validator.updateBeneficialOwnerAndManagingOfficerType, checkValidations, updateReviewIndividualManagingOfficer.post);
+
+router.route(config.UPDATE_REVIEW_BENEFICIAL_OWNER_GOV_URL + config.UPDATE_REVIEW_OWNERS_PARAMS)
```
Comment (stephanie-ch) [inline src/routes/index.ts:566]: UPDATE_REVIEW_BENEFICIAL_OWNER_GOV_URL doesn't look right?

File: src/controllers/update/update.review.beneficial.owner.gov.ts [hunk 7]
```diff
@@ -64,4 +64,3 @@ export const post = async (req: Request, res: Response, next: NextFunction) => {
     next(error);
   }
 };
-
```
Comment (corybudd-ch) [inline src/controllers/update/update.review.beneficial.owner.gov.ts:67]: could undo this change if you want, reduce the number of files changed

File: src/controllers/update/update.review.beneficial.owner.other.ts [hunk 8]
```diff
@@ -77,7 +77,6 @@ export const post = async (req: Request, res: Response, next: NextFunction) => {
 
     return res.redirect(UPDATE_BENEFICIAL_OWNER_TYPE_URL);
   } catch (error) {
-    console.log(error);
```
Comment (corybudd-ch) [inline src/controllers/update/update.review.beneficial.owner.other.ts:80]: 😱

File: src/routes/index.ts [hunk 9]
```diff
@@ -443,7 +443,13 @@ router.route(config.UPDATE_REVIEW_BENEFICIAL_OWNER_OTHER_URL)
   )
```
Comment (corybudd-ch) [inline src/routes/index.ts:443]: should you have changes in this file? Might be leftover from UAR-151 before it was merged. I'd just undo all these changes unless you're adding a new page in your ticket

File: views/update/update-review-individual-managing-officer.html [hunk 10]
```diff
@@ -12,7 +12,6 @@
   {% include "includes/back-link.html" %}
 {% endblock %}
 
-
```
Comment (corybudd-ch) [inline views/update/update-review-individual-managing-officer.html:15]: can remove this as well if you want less file changes

File: src/routes/index.ts [hunk 11]
```diff
@@ -533,6 +554,22 @@ router.route(config.UPDATE_MANAGING_OFFICER_URL + config.ID)
   .post(...validator.updateManagingOfficerIndividual, checkValidations, updateManagingOfficerIndividual.update);
 router.get(config.UPDATE_MANAGING_OFFICER_URL + config.REMOVE + config.ID, authentication, navigation.hasUpdatePresenter, updateManagingOfficerIndividual.remove);
 
+router.route(config.UPDATE_REVIEW_INDIVIDUAL_MANAGING_OFFICER_URL)
```
Comment (stephanie-ch) [inline src/routes/index.ts:557]: I don't think you need this either, looks defined above. Probably a merge issue :)

File: src/routes/index.ts (hunk 6 above)
Comment (stephanie-ch) [reply src/routes/index.ts:566]: I think this is defined above already

**Review by corybudd-ch — APPROVED** [review]

File: src/controllers/update/update.review.beneficial.owner.other.ts (hunk 8 above)
Comment (corybudd-ch) [reply src/controllers/update/update.review.beneficial.owner.other.ts:80]: no definitely keep, shouldn't have console.logs in there, my reaction was to its mere presence in the first place 😆

File: src/routes/index.ts (hunk 9 above)
Comment (cdrainCH) [reply src/routes/index.ts:443]: This is to populate the service address fields on page load, although I appreciate this may be out of scope for my ticket. Can remove and take note for future fix?

File: src/controllers/update/beneficial.owner.type.controller.ts (hunk 5 above)
Comment (cdrainCH) [reply src/controllers/update/beneficial.owner.type.controller.ts]: Removed

File: src/routes/index.ts (hunk 6 above)
Comment (cdrainCH) [reply src/routes/index.ts]: Removed

File: src/routes/index.ts (hunk 11 above)
Comment (cdrainCH) [reply src/routes/index.ts]: Removed
